x-pack/plugins/endpoint/server/services/resolver/query_builder.ts

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { isLegacyEntityID, parseLegacyEntityID, CountQueryInfo } from './common';
+import { EndpointAppContext, JSONish } from '../../types';
+import { EndpointAppConstants } from '../../../common/types';
+
+export interface PaginationInfo {
+  page: number | undefined;
+  pageSize: number | undefined;
+}
+
+function buildLegacyChildrenQuery(endpointID: string, uniquePID: string) {
+  return {
+    bool: {
+      filter: {
+        bool: {
+          should: [
+            {
+              bool: {
+                filter: [
+                  {
+                    term: { 'endgame.unique_pid': uniquePID },
+                  },
+                  {
+                    // had to change these to match queries otherwise I can't get results back, using the kibana
+                    // developer mode it works fine AND the api tests when it's also a term...not sure why
+                    match: { 'agent.id': endpointID },
+                  },
+                ],
+              },
+            },
+            {
+              bool: {
+                filter: [
+                  {
+                    term: { 'endgame.unique_ppid': uniquePID },
+                  },
+                  {
+                    match: { 'agent.id': endpointID },
+                  },
+                ],
+              },
+            },
+          ],
+        },
+      },
+    },
+  };
+}
+
+function buildESEndpointChildrenQuery(entityID: string) {
+  return {
+    bool: {
+      filter: {
+        bool: {
+          should: [
+            {
+              match: { 'endpoint.process.entity_id': entityID },
```

Comment:
By including `match: { 'endpoint.process.entity_id': entityID }` here, this query will return a superset of the documents returned by `buildESEndpointNodeQuery`.  As far as I can tell the response from the resolver/children endpoint is also a superset of the information provided by the resolver/node endpoint.  I don't think we want to return the detailed node information from both endpoints.